Block: social links.

v9ai opened the pull request:
## Description

Fixes: https://github.com/WordPress/gutenberg/issues/1873

## References
[slack discussion](https://wordpress.slack.com/archives/C02QB2JS7/p1553901163077800)
[Prototype](https://www.figma.com/proto/gql7fzWm5Ch401zkrLnBgvqU/social-icons-block-i2?node-id=0%3A2040&scaling=min-zoom)

## How has this been tested?

- Add new Social Links block
- Confirm you can insert multiple links
- Icons and links show on view
- Can create and use block as Author permission

## Screenshots 

(see updates in thread)


## Types of changes

Adds a new block for inserting social media links. The block will try to match the proper icon to the URL typed in, or the user can select the icon by clicking and selecting from a list.


## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

v9ai committed:
Added social links.

nicolad commented:
Hello @mkaz, thank you for your comments on the previous PR.
I opened this PR to make it possible to combine our efforts. Please let me know what is the branch name you are working on, so I can improve this block based on that and then have a better starting point.

mkaz commented:
I made a fork off your old code base and created a PR to show the difference.
https://github.com/nicolad/gutenberg/pull/1

If you like I can update and push that to this branch, I wanted you to review before I combined. It is still a work in progress, we need some clarification on what icon set to use. 

I like your idea of filling in the icon automatically based on the URL. I think we might be able to do both, if no icon is picked and the user types `twitter.com...` than the twitter icon shows; but also allow the user to click and set an icon if not found or they want to change.

nicolad commented:
> I made a fork off your old code base and created a PR to show the difference.
> [nicolad#1](https://github.com/nicolad/gutenberg/pull/1)
> 
> If you like I can update and push that to this branch, I wanted you to review before I combined. It is still a work in progress, we need some clarification on what icon set to use.
> 
> I like your idea of filling in the icon automatically based on the URL. I think we might be able to do both, if no icon is picked and the user types `twitter.com...` than the twitter icon shows; but also allow the user to click and set an icon if not found or they want to change.

Thank you for the help @mkaz, I will check and apply those changes to the current PR.

mkaz commented:
Also, I was just pointed to our set of social icons we can use here: 
https://github.com/Automattic/social-logos

v9ai committed:
Applied changes suggested by Marcus Kazmierczak.

nicolad commented:
> Also, I was just pointed to our set of social icons we can use here:
> https://github.com/Automattic/social-logos

I saw that in the prototype we have 2 styles, but `social-logos` don't have that option available(at least I didn't saw this being exposed via `props`).
![image](https://user-images.githubusercontent.com/1477453/62490728-b9d0e100-b7d2-11e9-8ff2-dff88401690f.png)

mkaz commented:
@nicolad Yeh, I saw that. Styles is an extra piece I think we can add on, once we get the link builder and picker in place and working well. I think we can add additional enhancements, also choosing color or matching theme, circles, etc...

mkaz committed:
Add social list of icons, SVGs in stylsheets

Initially add SVG as background images to get UX down.  We probably will
want to use real SVGs if possible to give more control over the color,
fill, stroke, etc...

Add listing of icons and associated matching domain/text. Automatically
check if URL entered matches and select icon.

mkaz commented:
I just pushed up a set of changes which should be workable. There are still a lot of polish and refinements to be made to the block, but general UI should be in a testable set.

mkaz committed:
Add fixture files

mkaz committed:
Alignment cleanup

mkaz committed:
Add icon picker, to pick icon

mkaz commented:
I think this is at a pretty good point for a design review. The interaction with InnerBlocks and hiding and showing based on selection is still not quite correct, but I'm not sure what direction to go.

@mapk, @melchoyce, and/or @karmatosed you all commented on the previous PR and issue. So if you are still interested, or any other designer input is welcome.

gziolo commented:
@matias, @youknowriad and @mapk - how much individual Social Link block and Logo block (https://github.com/WordPress/gutenberg/issues/16484) differ conceptually? I guess it's fine to proceed as two independent items, but this is another case where we should be thinking about having a block which depending on context could behave slightly different. I have no idea yet how this could be materialized but definitely something we should thing about :)

gziolo commented on packages/block-library/src/index.js:
```diff
@@ -59,8 +59,9 @@ import * as textColumns from './text-columns';
-
 import * as classic from './classic';
+import * as socialLinks from './social-links';
+import * as socialLink from './social-links/social-link';
```
Can we have Social Link block in the top-level folder as well to make their discovery easier?

mkaz replied:
My idea was to keep the social-link block as a child of social-links, so you wouldn't insert a single social-link but use them only as the inner blocks for social-links block.

There are a couple of different ways to solve, we could also just have a single social-link block, with no parent block, and then they are inserted multiple times using columns to control layout.

I felt that having the block with multiple fits more in line with how people use social links as a distinct block with multiple items.

I'm open to any suggestions here.

gziolo replied:
It's what we have at the moment for
- `columns` + `columns`
- `navigation-menu` + `navigation-menu-itme`

See https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src.

It's mostly for consistency and to have an easy way to discover all blocks with tooling. This way you can easily identify all `block.json` files with a simple pattern match.

mkaz replied:
:+1: Moved in 5668a9a56

gziolo commented on packages/block-library/src/social-links/social-link/edit.js:
```diff
@@ -0,0 +1,72 @@
+const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
+	const { icon, url } = attributes;
+
+	const classes = classNames( 'wp-social-icon', `wp-social-icon-${ icon }` );
```
Is this something we want to maintain on WordPress core scope from now on? At the moment corresponding styles will be loaded only together with the block.

mkaz replied:
I'm still not sure I understand this, this is how I would implement in a plugin, not quite sure how we should for core. Open to any suggestions

gziolo commented on packages/block-library/src/social-links/social-link/icon-picker.js:
```diff
@@ -0,0 +1,31 @@
+							key={ icon }
+							className={ classes }
+							data-icon={ icon }
+							onClick={ ( evt ) => onClick( evt.target.dataset.icon ) }
```
Is `data-icon` really necessary here, you can pass the name of the icon directly to the `onClick` handler?

mkaz replied:
I'll try to refactor, I was struggling passing it in and need to circle back on this one.

mkaz replied:
Fixed in 608dfaff0
I'm really not sure what I was doing the first time.

gziolo commented on packages/block-library/src/social-links/social-link/index.js:
```diff
@@ -0,0 +1,30 @@
+export const settings = {
+	title: __( 'Social link' ),
+	parent: [ 'core/social-links' ],
+	icon: 'share',
```
`parent` and `icon` should be moved to `block.json` file.

mkaz replied:
:+1:

gziolo commented on packages/block-library/src/social-links/index.js:
```diff
@@ -0,0 +1,23 @@
+
+export const settings = {
+	title: __( 'Social links' ),
+	icon: 'share',
```
`icon` should be moved to `block.json` file.

mkaz replied:
:+1:

melchoyce commented:
@gziolo Any chance I could get you to take a screen capture of the block for reference?

mkaz committed:
Move attributes to block.json

paaljoachim commented:
I do believe an icons block https://github.com/WordPress/gutenberg/issues/16484 could become a base for adding social and logo icons. As one could just add the icons block and add social icons with links or logo icons with links to produce something very similar to each other. 

That could also mean build out the icons block first and then build the social icons block using the elements from the icons block.

mkaz committed:
Move singular social-link to top-level

mkaz committed:
Pass parameter, dont use data-icon

I went done the wrong path, and took far too long to realize it.

mkaz commented:
@paaljoachim This is to some degree how I have it. I just refactored it in 5668a9a moving the singular `social-link` block out to its own block. The plural `social-links` block is a simple InnerBlocks allowing for only the `social-link` type.

The implementation is relatively straight-forward, but the interface isn't great using InnerBlocks with only a single block type. I tried to improve by using a custom appender, but not sure it's much better

mkaz commented:
> @gziolo Any chance I could get you to take a screen capture of the block for reference?

@melchoyce I included a GIF at the top which includes the current insert and edit interface.

mkaz committed:
Move social-link CSS to own scss

mkaz committed:
Switch inner-blocks to flex to inline

mkaz commented:
Here's a screen capture of the latest interface, improved but still can you some polish

![wp-social-links-rev-63](https://user-images.githubusercontent.com/45363/63073179-6fd7b180-bedc-11e9-81a5-404a7e309707.gif)

mkaz committed:
Add social inks to block transforms e2e tests

jasmussen commented:
Dig this project. I've worked a ton on social icons in the past, and have some thoughts, so I'll likely jump into this branch some time tomorrow and see if I can help!

jasmussen commented:
Alright, as a first step, here's a mockup of what I would love to see. I'm intentionally looking SLIIIGHTLY ahead, as to where we want to be. Then we'll figure out the smallest steps we can take to get there. 

Here's what you see immediately when you insert the Social Links block:

<img width="1282" alt="Social Links placeholder" src="https://user-images.githubusercontent.com/1204802/63423625-02141400-c40d-11e9-9b5f-ea40385665dc.png">

Let's break it down:

- `n` social links are already inserted, in this case 3
- We are already curators of what social networks go in WordPress, by virtue of providing an oembed whitelist. We are extending that to choose what those `n` preset social links are. In this case I randomly chose Facebook, Twitter, Linkedin, but I expect someone else than me to provide the number and type of links to preinsert here.
- Those three links are semi-transparent to indicate they are not filled in. 
- They are real "child blocks", as they are in this PR.
- Because none of those placeholder/preset blocks are "filled in", they are visible in their semi-transparent state, even when the block is not selected.
- When you click the Plus/Add Block button, the Block Library opens as a flyout showing all the other social links you can insert. This isn't a customized icon inserter, this is most literally the Block Library showing Child Blocks, one for each social link.

Okay, let's click the LinkedIn button:

<img width="1282" alt="Social Links, LinkedIn selected" src="https://user-images.githubusercontent.com/1204802/63423888-85ce0080-c40d-11e9-9fdd-ec9f41069d89.png">

Whaaa? Where are the clunky border outlines for _child blocks_? First off, go read this idea by @mtias: https://github.com/WordPress/gutenberg/issues/16974#issuecomment-520221183 — the pertinent bit:

> One idea I have for simplifying toolbar handling in nested contexts is to optionally allow an inner block container to say that it wants to absorb the toolbar of all the children. In those cases, selecting a child block within that area would change the toolbar content but it would be positioned at the level of the parent. This could be really useful for the navigation menu block, as selecting different items would remain coherent while offering specific tools for the right contexts.

Let's break it down:

- Each child block is passed a prop, say `minimalUI`, which minimizes the block UI that would otherwise be clunky for such a small thing.
- With minimal-UI, the block toolbar for the LinkedIn block is absorbed by the parent block, in this case, the Social Links block. 
- With minimal-UI, there aren't extra margins or paddings on the child block, or extra borders. It appears as though the parent block is selected, even though technically you've selected a child.
- When clicking the LinkedIn block, the profile URL immediately pops out in a popover. This is similar to a still-unmerged PR that intends to do the same for the Button block. 
- As soon as a single social link block is filled out, and you deselect the Social Links block, you only see that link. The semitransparent blocks become invisible until you select the block again. 
- _There's no mover_.*1 

***1: No mover?** Then how do you rearrange the blocks? Well, for now maybe you don't. Maybe you delete and insert in the right order.

Or, maybe there are options for "move left" and "move right" in the ellipsis menu, here:

<img width="251" alt="Screenshot 2019-08-21 at 12 24 41" src="https://user-images.githubusercontent.com/1204802/63424443-bbbfb480-c40e-11e9-874c-50b4ea9964eb.png">

And, maybe #17088 will surface some additional options for reordering these things which will benefit _all_ child blocks regardless of UI.

---

Okay, that was a lot. So how do we get from this PR to there? Well, we know we want this:

1. Child blocks
2. Icons
3. Curated colors on a per-icon basis
4. Let's pre-insert `n` buttons as soon as you insert the Social Links block.
5. We want to use the actual block library for the "Add block" button, and show actual child blocks in the block library. Not a custom button. See the Jetpack Contact form, where when that form is selected, the block library shows a filtered view of form widgets that can be inserted just in that block.
6. Let's make an un-filled child block have a semi transparent icon, and be opaque when created.
7. We want to push a little CSS polish to make sure the child blocks align just right, have the style and so on. 
8. Pending further discussions on the "minimal-UI" idea, we might want to create a new ticket for that, and in the mean time implement some CSS to get closer to that minimal UI just for this block. 

1 and 2 are already in a good place here. 3 and 4 I'm sure will spawn interesting discussions, but that can start now. @karmatosed @mapk do you have feelings about this? Which social link blocks do you think should be preinserted and how many?

5, 6 is code work, that can start now. 

7 and 8 I will start on as soon as I can: I imagine that just removing the block border, hiding the movers and nullifying the margins and paddings, can get us quite a ways there. We'll include in the code a link to the ticket for implementing minimal UI. 

Okay your turn: lay your thoughts on this approach.

mtias commented:
Great summary, I think it generally covers all the points. Some small feedback:

> Because none of those placeholder/preset blocks are "filled in", they are visible in their semi-transparent state, even when the block is not selected.

This worries me a bit because not the non-selected state doesn't reflect the real site. I don't want to see a permanent Facebook logo, even if semi-transparent, if I haven't opted in to it. I can imagine users being confused about it. If it shows up semi-transparent on selection, that seems fine and useful.

> There's no mover.*1

If we use child blocks here, then we'll inherit from #16637.

> Each child block is passed a prop, say minimalUI, which minimizes the block UI that would otherwise be clunky for such a small thing.

My thinking was that this would be the task of the `InnerBlocks` context and not the children. Something like `constrainToolbar`. Also related: #7469.

jasmussen commented:
Excellent thoughts, thanks for chiming in.

> This worries me a bit because not the non-selected state doesn't reflect the real site. I don't want to see a permanent Facebook logo, even if semi-transparent, if I haven't opted in to it. I can imagine users being confused about it. If it shows up semi-transparent on selection, that seems fine and useful.

I hear you. I think this is a key issue to think about, though, in order to make this block intuitive to use. As I've outlined the behavior, it is actually the same as that of the image block: you get placeholder content visible even when unselected, until you customize it.

Perhaps a different "disabled state" might help?

> If we use child blocks here, then we'll inherit from #16637.

Wonderful, yes. So maybe hide them until that pr unsticks? I'll also see if I can provide a mockup to share in that thread.

> My thinking was that this would be the task of the InnerBlocks context and not the children. Something like constrainToolbar

Ah right, that was even clear rereading the comment in quoted. The visual really and effect might be the same, right?

mtias commented:
> you get placeholder content visible even when unselected, until you customize it.

I'd be ok with this being the default state when you add the block and unselect it. But as soon as I add a single icon, the non-selected state should reflect what I see on the front. If I only configure an instagram logo, I'd expect only that to be shown when the block is not selected.

> Ah right, that was even clear rereading the comment in quoted. The visual style and effect might be the same, right?

Possibly, though there might be two different needs at play here:

* Absorb the toolbar controls in the place of the parent toolbar, eliminating the block UI of the children.
* Eliminate the block UI of a container.
* A mixture of both.

jasmussen commented:
> I'd be ok with this being the default state when you add the block and unselect it. But as soon as I add a single icon, the non-selected state should reflect what I see on the front. If I only configure an instagram logo, I'd expect only that to be shown when the block is not selected.

Yes indeed! That would be the full behavior, I was not clear enough. 

- Visible when inserted fresh and unselected.
- As soon as customized in any way, non-selected state is same as frontend. 

> Absorb the toolbar controls in the place of the parent toolbar, eliminating the block UI of the children.
> Eliminate the block UI of a container.
> A mixture of both.

Also related, perhaps: #7694. 

I definitely think that what you're referring to here, minimal-UI or whatever we might call it, would need some time in the oven, but could potentially start with the idea more or less outlined by our needs here.

mkaz committed:
Show icons when block inserted, click to select

Removes switching icon based on URL typed because someone is
selecting the icon first. This makes it harder to change an icon
after the fact, will require delete and insert.

mkaz committed:
Improve inserting slightly

- Remove text link, just use icon button (+)

- Start with template, so you dont have to click (+) after inserting
block, you immediately select the icon

mkaz committed:
Display URLInput Popover on insert

Switches URLInput field to Popover so it can be controlled to display
on insert. This allows the user to know they have to fill in the URL for
the icon and removes an extra click on initial insert to set.

Commit:
Make progress towards minimal UI.

- This removes margins, paddings, and lots of UI from child blocks. It also adds comments that we should look at refactoring this away in a separate PR that makes it a prop on a parent container.
- This tweaks the layout a little bit, so frontend and backend match, and adds a circle around the social icons.

jasmussen commented:
I pushed a change to make some progress on the suggestions outlined in https://github.com/WordPress/gutenberg/pull/16897#issuecomment-523399945:

![progress](https://user-images.githubusercontent.com/1204802/63501602-326eb780-c4cc-11e9-94ce-093e75a03bcb.gif)

Frontend:

<img width="558" alt="Screenshot 2019-08-22 at 10 56 28" src="https://user-images.githubusercontent.com/1204802/63501607-37336b80-c4cc-11e9-9e05-367e47565641.png">

There's some work around the appender and default state that needs work, I'll share some mockups around that in a bit, but first I'll explore how we can colorize the SVGs. I notice that the SVGs are all inserted as background images, which is not ideal. We could use something like https://codepen.io/noahblon/post/coloring-svgs-in-css-background-images, but that doesn't support IE11, but suffice to say I'll look into this.

mkaz committed:
Refactor Social Links to create individual blocks

Creates individual blocks for each social site. This uses the standard
InnerBlocks appender and display to show icon/site to select. Not only
does this use a standard widget, it benefits by getting search
capabilities.

Modeled after the embed include, however each of the sites are not a
"public" block shown in the inserter, but only seen when using the
Social Links block. There would be confusion using the inserter with
two entries for example adding a link to Twitter vs. embedding a Tweet

mkaz commented:
@jasmussen I pushed up a change set which gets us close to what we want. Everything was switched over to create individual blocks for each social site, and then uses the standard InnerBlocks appender to insert.

There are still two pieces remaining, not counting the various polish and testing.

- [ ] Use not have a URL set as the "faded" case, I'll update to add a class to the link if a URL is not set, so you can add some magic to fade it. Also, I'll update save to not save an icon if a URL is not set.

- [ ] Icons are still an issue. You can see in the inserter when adding a child block, they are using the "share" icon. The editor part is easier to solve, since we can inline SVGs there. However, saving inline SVG in the block still runs into kses issues.

mkaz committed:
Add is-incomplete class for empty URL

- Adds wp-social-icon__is-incomplete class to social icon
block if the URL is not set.

- Checks for URL on save and does not include icon if not set

mkaz commented:
Added `wp-social-icon__is-incomplete` if the URL is not set. I added a basic opacity, feel free to tweak. I also updated the save so an empty URL won't save.

mkaz committed:
Show default icon, set icon attribute fixes save

jasmussen commented:
Very nice work.

I noticed that when you open the block library, there are three sections: social links, most used, and "Widgets", and they all contain the same social blocks. Can we have just one section? Not sure what our options are here.

Commit:
Polish appender and inactive state.

mtias commented on packages/block-library/src/social-link/save.js:
```diff
@@ -0,0 +1,23 @@
+		return null;
+	}
+
+	const classes = classNames( 'wp-social-icon', `wp-social-icon-${ icon }` );
```
Sometimes this is referred to as `social-links` and sometimes as `social-icon`.

mkaz replied:
Fixed, standardized everything on social-link name

mtias commented on packages/block-library/src/social-link/save.js:
```diff
@@ -0,0 +1,23 @@
+
+	const classes = classNames( 'wp-social-icon', `wp-social-icon-${ icon }` );
+	return (
+		<span className={ className }>
```
Can we have more meaningful markup?

mkaz replied:
Switched to list tags

mtias commented on packages/block-library/src/social-links/edit.js:
```diff
@@ -0,0 +1,40 @@
+
+export const SocialLinksEdit = function( { className } ) {
+	return (
+		<>
```
These fragment seems unnecessary.

mkaz replied:
Removed.

mkaz committed:
Remove unnecessary fragment

mkaz committed:
Convert icons over to React components, show in inserter

mkaz committed:
Clean up

- standardize on social-link naming, fix CSS
- use icon attribute set as default in register

mkaz committed:
Tidy things up, rename, consistent

- Update icon names to component case
- Use IconComponents in edit view
- Updates variables so site is for sites, icon is for icons ;-)
- Switch to use <ul> and <li> for markup

mkaz commented:
A set of changes went in yesterday which refactors all the icons so they can be inline and adjustable via CSS. It uses server-side render for displaying the icons on save, and React components for the icons in the inserter and editor. They don't currently have any extra styling, but we can create attributes if we want to trigger fill colors or other things to pass in to the SVG.

> I noticed that when you open the block library, there are three sections: social links, most used, and "Widgets", and they all contain the same social blocks. Can we have just one section? Not sure what our options are here.

What is shown is all automatic with the InnerBlocks Inserter, I don't know of a way to customize without creating our own appender and subsequent UI.

A few remaining items:

- [ ] Hide icons not complete when the block is not selected. The state when viewing in the editor, but not editing the block (isSelected), should resemble the published state.

- [ ] Improve URL input component, the current url input field should probably resemble the inline url input field, that includes a return-arrow after the text and closes when typing enter.

- [ ] Get the tests passing again ;-)

Commit:
Size icons correctly, front and back. Add focus style.

Commit:
Add colors.

jasmussen commented:
Okay, I pushed a little fix so the SVGs are correctly positioned, front and backend, and added a focusring:

<img width="419" alt="Screenshot 2019-08-26 at 11 47 34" src="https://user-images.githubusercontent.com/1204802/63682045-5812fe00-c7f7-11e9-9698-e37dde9b59f2.png">

In a slog of work, I also managed to add colors to each icon. Here's a sampling:

<img width="624" alt="Screenshot 2019-08-26 at 12 47 40" src="https://user-images.githubusercontent.com/1204802/63685597-1044a480-c800-11e9-94e7-8f4743be92e0.png">

<img width="533" alt="Screenshot 2019-08-26 at 12 47 42" src="https://user-images.githubusercontent.com/1204802/63685599-1175d180-c800-11e9-89f7-db378c34e3c3.png">

With this, I think we're close. Just need that tasklist you mentioned, as well as creating a new ticket for the minimal UI prop.

jasmussen commented:
@mkaz Where did you get the list of icons for?

In a previous commit, I created the following CSS for colorizing:

```
.wp-social-link-amazon {
	background-color: #f90;
	color: #fff;
}

.wp-social-link-apple {
	background-color: #f1f1f1;
	color: #000;
}

.wp-social-link-bandcamp {
	background-color: #1ea0c3;
	color: #fff;
}

.wp-social-link-behance {
	background-color: #0757fe;
	color: #fff;
}

.wp-social-link-chain {
	// Link, has generic colors.
}

.wp-social-link-codepen {
	background-color: #1e1f26;
	color: #fff;
}

.wp-social-link-deviantart {
	background-color: #02e49b;
	color: #fff;
}

.wp-social-link-digg {
	background-color: #f1f1f1;
	color: #1b1a19;
}

.wp-social-link-dribbble {
	background-color: #e94c89;
	color: #fff;
}

.wp-social-link-dropbox {
	background-color: #4280ff;
	color: #fff;
}

.wp-social-link-etsy {
	background-color: #f45800;
	color: #fff;
}

.wp-social-link-facebook {
	background-color: #1977f2;
	color: #fff;
}

.wp-social-link-feed {
	// Feed, has generic colors.
}

.wp-social-link-fivehundredpx {
	background-color: #000;
	color: #fff;
}

.wp-social-link-flickr {
	background-color: #0461dd;
	color: #fff;
}

.wp-social-link-foursquare {
	background-color: #e65678;
	color: #fff;
}

.wp-social-link-github {
	background-color: #24292d;
	color: #fff;
}

.wp-social-link-goodreads {
	background-color: #eceadd;
	color: #382110;
}

.wp-social-link-google {
	background-color: #ea4434;
	color: #fff;
}

.wp-social-link-instagram {
	background-color: #f1f1f1;
	color: #000;
}

.wp-social-link-lastfm {
	background-color: #e21b24;
	color: #fff;
}

.wp-social-link-linkedin {
	background-color: #0577b5;
	color: #fff;
}

.wp-social-link-mail {
	// Mail, has generic colors.
}

.wp-social-link-mastodon {
	background-color: #3288d4;
	color: #fff;
}

.wp-social-link-medium {
	background-color: #02ab6c;
	color: #fff;
}

.wp-social-link-meetup {
	background-color: #f6405f;
	color: #fff;
}

.wp-social-link-pinterest {
	background-color: #e60122;
	color: #fff;
}

.wp-social-link-pocket {
	background-color: #ef4155;
	color: #fff;
}

.wp-social-link-reddit {
	background-color: #fe4500;
	color: #fff;
}

.wp-social-link-skype {
	background-color: #0478d7;
	color: #fff;
}

.wp-social-link-slideshare {
	background-color: #0577b5;
	color: #fff;
}

.wp-social-link-snapchat {
	background-color: #fefc00;
	color: #050708;
}

.wp-social-link-soundcloud {
	background-color: #ff5600;
	color: #fff;
}

.wp-social-link-spotify {
	background-color: #1bd760;
	color: #fff;
}

.wp-social-link-stumbleupon {
	background-color: #eb4923;
	color: #fff;
}

.wp-social-link-tumblr {
	background-color: #011835;
	color: #fff;
}

.wp-social-link-twitch {
	background-color: #6440a4;
	color: #fff;
}

.wp-social-link-twitter {
	background-color: #21a1f3;
	color: #fff;
}

.wp-social-link-vimeo {
	background-color: #1eb7ea;
	color: #fff;
}

.wp-social-link-vk {
	background-color: #4680c2;
	color: #fff;
}

.wp-social-link-wordpress {
	background-color: #3499cd;
	color: #fff;
}

.wp-social-link-yelp {
	background-color: #d32422;
	color: #fff;
}

.wp-social-link-youtube {
	background-color: #ff0100;
	color: #fff;
}
```

That's a lot of links, and I'm personally of the idea that "the fewer the better". But I'm wondering where the list is from — in the case of embeds, it's the list of oembeds WordPress already supports. But this one seems a little random, for example I'm not even sure what to type in the "Apple" list. Can we try and trim this down a little bit?

jasmussen committed:
Add style variation for icon only.

jasmussen commented:
I added a style variation for icon only:

![Screenshot 2019-08-26 at 15 18 54](https://user-images.githubusercontent.com/1204802/63694166-1c3b6100-c816-11e9-89a3-557f851ea37b.png)

Good for when you want to insert this in a menu, that sort of thing.

Note that the preview is a little funky, I think we'll want a good solid rebase of this branch.

jasmussen commented:
What's "Index View"?

We can never use !important ;) — troll Marcus. But I'll consider that a good test and bug report. I can fix both things tomorrow. Same with the big ones, we'll probably want to add explicit width/heights to the SVGs. 

I don't know why the tests fail, though, and that "hide uncustomized icons" I could use your skill with 🧡

mkaz committed:
Hide incomplete link if parent or child not selected

jasmussen commented:
> I added a smidge of CSS to hide when parent or child is not selected, should now behave like expected in edit view.

![has-incomplete](https://user-images.githubusercontent.com/1204802/63751860-77219680-c8b0-11e9-9aa4-e147dff6455c.gif)

I think it needs a little more love. I realize this is some complexity, and if it's too much, then we can rethink the approach. But basically it should behave like the following matrix:

- All blocks have URLs: all blocks show full opacity with parent block selected or unselected ✅
- Some blocks have URLs: only those with URLs show in full opacity with parent block selected, blocks without URLs disappear when parent block is unselected, and are semitransparent when parent block selected ✅
- No blocks have URLs: all placeholder blocks appear semitransparent even when the parent block is unselected. 👈 

The 3rd one is missing. It's also the hardest to explain. The idea is that if you have just inserted the social links block, it is a placeholder, and therefore shows up even when unselected. Just like the Image block.

jasmussen committed:
Remove a couple of icons. Comment out for now.

jasmussen commented:
Just to force the discussion, I tried removing some icons. This is the list now:

```
	AmazonIcon,
	//AppleIcon,
	BandcampIcon,
	BehanceIcon,
	ChainIcon,
	CodepenIcon,
	DeviantartIcon,
	//DiggIcon,
	DribbbleIcon,
	//DropboxIcon,
	EtsyIcon,
	FacebookIcon,
	FeedIcon,
	FivehundredpxIcon,
	FlickrIcon,
	FoursquareIcon,
	GoodreadsIcon,
	//GoogleIcon,
	GithubIcon,
	InstagramIcon,
	LastfmIcon,
	LinkedinIcon,
	MailIcon,
	MastodonIcon,
	//MeetupIcon,
	MediumIcon,
	PinterestIcon,
	PocketIcon,
	RedditIcon,
	SkypeIcon,
	//SlideshareIcon,
	SnapchatIcon,
	SoundcloudIcon,
	SpotifyIcon,
	//StumbleuponIcon,
	TumblrIcon,
	TwitchIcon,
	TwitterIcon,
	VimeoIcon,
	VkIcon,
	WordPressIcon,
	//YelpIcon,
	YoutubeIcon,
```

- Stumbleupon doesn't exist anymore. 
- Neither does Slideshare
- Why would you link to Meetup in a social links block?
- Google Plus doesn't exist in a public facing form.
- What Dropbox URL would you link to? 
- Same with Apple. 
- Digg isn't really for sharing and digging anymore, where in the past these links would be to submit the URL.

Given the purpose of this block is most literally, Social Links, it feels like it's important to take out anything that isn't web-social. 

But I have only commented out for now, I'm sure people will have opinions. Once we settle the debate, we can actually remove the files.

karmatosed commented:
Just testing this, I am not sure the 'search' for links is working. Or at least it doesn't seem to be for me. It made me wonder, do you want to search as this is about linking to URL?

![image](https://user-images.githubusercontent.com/253067/63765951-0daa8380-c8c2-11e9-9118-666aef8375f2.png)

karmatosed commented:
Another thing that struck me was the block library could maybe do with some adaption. For example, not showing the Tip copy/content block. I also wonder if 'searching blocks' makes sense here?

![image](https://user-images.githubusercontent.com/253067/63766027-392d6e00-c8c2-11e9-8bf6-a9ef2236cf97.png)

mkaz committed:
Improve URLInpout, switch to URLPopover

mkaz committed:
Update and fix tests, get to green

- Add tests for individual social link blocks
- Generate fixtures
- Add block transforms (Group)
- PHP has lint rules too

Commit:
Update facebook icon and instagram colors.

jasmussen commented:
I updated thej facebook icon and changed the Instagram color:

<img width="209" alt="Screenshot 2019-08-28 at 10 51 53" src="https://user-images.githubusercontent.com/1204802/63840838-ff23a100-c981-11e9-8643-2cf2206108ae.png">

These are according to brand guidelines, both of them.

Commit:
Polish, add variation, bugfix

- This adds explicit widths heights to SVGs, fixing the "giant svg" issue.
- It adds colors to the now called "Logos Only" variation
- It improves the snapchat style
- It adds a pill variation

jasmussen commented:
Added a few things. Colors to logos only style (changed the name from icon only):

<img width="372" alt="Screenshot 2019-08-28 at 11 21 36" src="https://user-images.githubusercontent.com/1204802/63844068-cab2e380-c987-11e9-9601-4ecca8b1d5cc.png">

Improved snapchat style:

<img width="117" alt="Screenshot 2019-08-28 at 11 22 33" src="https://user-images.githubusercontent.com/1204802/63844081-d0a8c480-c987-11e9-9681-d75a133242e1.png">

There was also a typo in that name, "Snapshot" :D someone has been tirelessly working on tests I can tell!

New variation, pill shape:

<img width="378" alt="Screenshot 2019-08-28 at 11 29 48" src="https://user-images.githubusercontent.com/1204802/63844113-e322fe00-c987-11e9-946c-0d5122aa3dd0.png">

Commit:
Polish, and actually remove icons.

This removes some comments.

It restores some icons, but it permanently removes others:

- Apple
- Slideshare
- Stumbleupon
- Digg

jasmussen commented:
In https://github.com/WordPress/gutenberg/pull/16897/commits/a11fea9e178e565e83d7a2a1c8ea0a4397276ce7 I removed these icons:

- Digg. This is no longer a tool where users can share their stories, as far as I can tell.
- Apple — what even is an Apple social profile
- Slideshare — retired, part of LinkedIn now
- Stumbleupon — also retired

Dropbox, Google, Meetup, and a few others were restored. It's fine with a little choice I guess.

Commit:
Fix URL popover.

jasmussen commented:
![links](https://user-images.githubusercontent.com/1204802/63845971-2af75480-c98b-11e9-9060-17139b318294.gif)

Fixed the URL popover.

For some reason, `<input type="text">` is used for the regular link popover, and is styled to look good, whereas the `url` type is not. Any damage in using `text`?

talldan commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,60 @@
+	const classes = classNames(
+		'wp-social-link',
+		'wp-social-link-' + site,
+		{ 'wp-social-link__is-incomplete': ( url ) ? false : true },
```
```suggestion
		{ 'wp-social-link__is-incomplete': url },
```

talldan commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,60 @@
+
+const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
+	const { url, site } = attributes;
+	const [ showURLPopover, setPopover ] = useState( true );
```
I found it unusual that this defaults to `true`. When tabbing into one of the child blocks the popover opens unexpectedly and tabbing is then constrained.

I understand that this is prompting the user to add a URL, but I don't think it's a good pattern from an accessibility point of view.

jasmussen replied:
Yep, I think this seems good. Thanks.

Commit:
This improves the preview.

jasmussen commented:
Improved the preview a bit:

<img width="481" alt="Screenshot 2019-08-28 at 12 14 31" src="https://user-images.githubusercontent.com/1204802/63847168-832f5600-c98d-11e9-907a-5eccffcb43b6.png">

Some of the CSS written in https://github.com/WordPress/gutenberg/pull/16897/commits/aaaee8c3e8bedf7eddfbd36785c320ba3858b7c1 might be redundant with #17064

talldan commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,60 @@
+						<div className="editor-url-input block-editor-url-input">
+							<input type="text"
+								value={ url }
+								onChange={ ( evt ) => setAttributes( { url: evt.target.value } ) }
```
Should avoid abbreviations where possible
```suggestion
								onChange={ ( event ) => setAttributes( { url: event.target.value } ) }
```

talldan commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,60 @@
+				<URLPopover
+					onClose={ () => setPopover( false ) }
+				>
+					<form className="block-editor-url-popover__link-editor" onSubmit={ () => setPopover( false ) }>
```
I think the `onSubmit` needs a `preventDefault` as I'm seeing a warning in the browser when pressing the apply button:
`Form submission canceled because the form is not connected`

talldan commented on packages/block-library/src/social-links/edit.js:
```diff
@@ -0,0 +1,38 @@
+	return 'core/social-link-' + site;
+} );
+
+// Template contains the links that show when start
```
```suggestion
// Template contains the links that show when start.
```

talldan commented on packages/block-library/src/social-links/edit.js:
```diff
@@ -0,0 +1,38 @@
+			<InnerBlocks
+				allowedBlocks={ ALLOWED_BLOCKS }
+				templateLock={ false }
+				template={ TEMPLATE }
```
I wonder if the larger button appender might be more useful than four social icons that the user might not even want. It would also solve the issue of the block being empty when unselected.

jasmussen replied:
I really like the idea of being a little bit more colorful, opinionated and even playful with this block, by preinserting some blocks. It immediately makes clear what this block is about, moreso than the gray appender, which could be anything, and often is.

However, let's bookmark this idea as a solid alternative for the empty state! Thanks Dan!

talldan commented on packages/block-library/src/social-link/index.js:
```diff
@@ -0,0 +1,48 @@
+import save from './save';
+import socialList from './social-list';
+
+const commonAttribs = {
```
Would be good to not use an abbreviation here as well.

talldan reviewed (commented):
Realised this is still WIP halfway through my review. Thought I'd still give some early feedback 😄

Commit:
Try a new "unselected state".

jasmussen commented:
Alright, in https://github.com/WordPress/gutenberg/pull/16897/commits/5891e55ae96ffd70fd728c5a49d122218d460399 I blazed ahead with a new approach to the placeholders. It now works like this:

![social links](https://user-images.githubusercontent.com/1204802/63854127-b4fbe900-c99c-11e9-8154-fc13bf8672ce.gif)

- It's performant and simple
- It's kinda fun
- An unselected block is a preview, always

The primary downside is that you might have a social links block inserted but with no blocks configured, and it will look as if it isn't there when the block is not selected. However, the min-height ensures that _something_ is there, and @mtias suggested we try and take a stab at this challenge in a different way:

> I think we can catch the first case by making FB and Twitter have a url set but without a profile
so they are always set

In other words — have one or two of the social link buttons be _not_ incomplete, even if the URL that is set is not the real one. I definitely think we should try this. @mkaz can you help with that?

mkaz committed:
Apply suggestions from code review

Thanks for taking the time and reviewing

Co-Authored-By: Daniel Richards <daniel.richards@automattic.com>

mkaz committed:
Touch up remaining user interactions

- Add WordPress as default logo, so does not collapse to nothing
- Per @talldan review, add preventDefault on submit
- Clean up removed icons
- Fix ! url logic

jasmussen commented:
Awesome, can't wait to see this land. 

> @talldan I agree the setting the Popover is a little off putting, but it was to show that there is a URL to edit for each icon, so helps with discoverability, but might not be worth it if it is an accessibility issue. @jasmussen any thoughts?

Yeah, so the Button block is actually the template for this, and it works like so:

![button](https://user-images.githubusercontent.com/1204802/63870745-dec40880-c9ba-11e9-8694-b1faf08e0450.gif)

The "Social Link" child block is also just a button with a URL being the primary thing you set. So these two should behave the same: Select Button, URL field pops out. Select Social Icon, URL field pops out.

I think the key here is, we don't set _focus_ in the URL field, you have to click that or tab there yourself. 

The styling here is also different from the Button block, but the behavior should still be the same. I recall an as-yet unmerged PR that changes the Button block to also have a popout URL field. I mean to dig that PR up, brush it off, and get it merged.

An additional benefit to this is, we can take this new Social Links block, when it lands, and use it as a template for a "Buttons" block, one that contains multiple individual button blocks.

jasmussen committed:
Hit area & frontend refinement.

jasmussen commented:
Okay, pushed a little additional polish. On the frontend, the hover color was overridden, and the hit area was small. This fixes that, and makes a little hover scale effect too:

![hover](https://user-images.githubusercontent.com/1204802/63873640-429d0000-c9c0-11e9-90cd-85161fb421c5.gif)

mkaz committed:
Remove abbreviation in variable name

mkaz committed:
Remove apple, digg, slideshare, stumbleupon from register

mkaz committed:
Add e2e tests for dropbox, google, meetup, yelp

mkaz committed:
Fix e2e-test case fixtures

mkaz committed:
Don't open URL popover by default, requires click in

jasmussen commented:
I'm liking this one a lot. I think it's definitely ready for final reviews by anyone who has time to test it.

jasmussen commented on packages/block-library/src/social-link/index.js:
```diff
@@ -0,0 +1,48 @@
+};
+
+// Create individual blocks out of each site in social-list.js
+// TODO: solve icon issue
```
@mkaz did we solve this?

jasmussen commented on packages/block-library/src/social-link/save.js:
```diff
@@ -0,0 +1,25 @@
+	}
+
+	// TODO: suggestions for fall back icon?
+	// @todo: These classes don't appear to render in the final markup at all.
```
Marcus, can you take a look? I think we might be able to remove both comments and the classname thing below.

jasmussen commented on packages/block-library/src/social-links/editor.scss:
```diff
@@ -0,0 +1,109 @@
+}
+
+// Reduce the paddings, margins, and UI of inner-blocks.
+// @todo: eventually we may add a feature that lets a parent container absorb the block UI of a child block.
```
I will make a ticket for this.

jasmussen committed:
Make icons in the "logos only" style bigger.

jasmussen commented:
I made the logos bigger, when they are in the "logos only" style:

![Screenshot 2019-08-29 at 12 21 02](https://user-images.githubusercontent.com/1204802/63932556-b4745880-ca57-11e9-803d-0a9097716c6e.png)

mkaz committed:
Remove save, using server-side render so not needed

jasmussen committed:
Comment polish, remove redundant rule, fix gray.

mkaz committed:
Remove TODO comment, icon issue solved

mkaz committed:
Update e2e tests with pure server-side rendering

apeatling commented:
Gave this a test, and I really love how simple and clear the interface is. Especially how each service is greyed out and doesn't show on the front end until you add a link.

The only thing I would say is I expected to be able to select an icon and hit the delete key to remove it, rather than having to navigate the menu.

jasmussen commented:
> The only thing I would say is I expected to be able to select an icon and hit the delete key to remove it, rather than having to navigate the menu.

Yes, this is a side effect of the _click_ action on a child block opening and setting focus on the URL field. If you press Escape to close the URL field, then you can press Delete.

If you tab through the block, you'll focus first the child block, then the URL field, and so on. 

An alternative to this would be for focus to remain on the child block, and not set focus in the URL field, not just for tabbing but for mouse too. @mkaz I believe you investigated this?

apeatling commented:
This makes sense. What if the delete action only applies to the child block if the URL field is empty? In this case there is no use for the delete key in the URL field since the field is blank. Also, if I'm deleting an icon there may be a high chance that URL field is already empty?

apeatling commented:
Thinking about this more, it would cause an issue with deleting a URL, and then hitting the delete key one too many times and accidentally deleting the icon.

jasmussen commented:
Created a ticket for making a generic prop for the minimal child block UI: #17267

jasmussen committed:
Polish. Remove obsolete code, add generic colors.

jasmussen commented:
Polished a little bit:

![links](https://user-images.githubusercontent.com/1204802/64015522-ff0fd680-cb24-11e9-9997-f438d002a7c8.gif)

Also, @mkaz it is, as Andy suggests, a little hard to delete a block. Would it be possible so making it so that clicking a social link selects it (so when you press delete it disappears), but the URL field _does_ pop out, it just doesn't have focus?

jasmussen committed:
Add centering.

mtias commented:
> but the URL field does pop out, it just doesn't have focus?

This is an accessibility affordance of the dropdown in general, I think, as opening it has to move the focus. What I think could be better would be moving the URL to the toolbar, since using child blocks now makes that a lot more ergonomic and scalable (particular for mobile). But we can explore this separately.

jasmussen commented:
Cool, I'll make some tickets. Here's a quick mockup:

![immediate](https://user-images.githubusercontent.com/1204802/64022214-fecc0700-cb35-11e9-9917-b08105a6bca6.png)

Also posted an alternate future version here: https://github.com/WordPress/gutenberg/issues/17267#issuecomment-526590324

jasmussen commented:
Okay as you can tell, I've now created a bunch of links for followup issues. These are all features/enhancements that can be explored separately, on their own cadence, and some of them likely need a little discussion first. 

- Additional flex properties for Social Links and Buttons alignments in inspector: https://github.com/WordPress/gutenberg/issues/17275
- Absorb child UI: https://github.com/WordPress/gutenberg/issues/17267
- Allow plugins to add new logos: https://github.com/WordPress/gutenberg/issues/17277
- Allow plugins to change social logos defaults: https://github.com/WordPress/gutenberg/issues/17278
- Add sizing controls to the icons: https://github.com/WordPress/gutenberg/issues/17276

With these ideas tracked, we ensure nothing falls off the map if we go ahead and review and hopefully merge this one ;)

mkaz commented:
I additionally created the ticket #17280 for customizing the InnerBlocks block inserter

nosolosw commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,65 @@
+							<input type="text"
+								value={ url }
+								onChange={ ( event ) => setAttributes( { url: event.target.value } ) }
+								placeholder={ __( 'Enter Address or Profile' ) }
```
I was testing this: 

![2019-09-06-11-06-42-address-or-profile](https://user-images.githubusercontent.com/583546/64418308-0a21b400-d09b-11e9-8f31-13ee69ed595f.png)

and my first thought when seeing the placeholder text was: "_ok, I can add the full URL (twitter.com/nosolosw) or my nickname(nosolosw)_". It turns out the nickname approach wasn't possible. So, how about simplifying this text to:

```suggestion
								placeholder={ __( 'Enter Address' ) }
```

jasmussen committed:
Update packages/block-library/src/social-link/edit.js

Dig it.

Co-Authored-By: Andrés <nosolosw@users.noreply.github.com>

nosolosw commented on packages/block-library/src/social-links/block.json:
```diff
@@ -0,0 +1,6 @@
+{
+	"name": "core/social-links",
+	"category": "widgets",
+	"icon": "share",
```
I see that `icon` is also declared in `social-links/index.js`. Do we need it in both places?

mkaz replied:
Good catch, removed the duplicate here.

nosolosw commented on packages/block-library/src/social-link/index.php:
```diff
@@ -0,0 +1,227 @@
+ * @return string Rendered HTML of the referenced block.
+ */
+function render_core_social_link( $attributes ) {
+	$site = ( isset( $attributes['site'] ) ) ? $attributes['site'] : 'Icon';
```
I was trying to understand in what situations the `site` att can be null, but I couldn't come up with any. I guess using the `share` icon and adding the `wp-social-link-Icon` is only a protective measure.

mkaz replied:
yeh, just a defensive measure, figure its better to be safe and not assume anything

oandregal reviewed (approved):
Code-wise this is :+1:

mkaz committed:
Remove duplicate icon definition

aduth commented on packages/block-library/src/social-link/edit.js:
```diff
@@ -0,0 +1,65 @@
+							event.preventDefault();
+							setPopover( false );
+						} } >
+						<div className="editor-url-input block-editor-url-input">
```
Class names are not meant to be reused outside the component which defines them. In this case, we should go through the abstraction of the `URLInput` component, not use its classes directly. Or, alternatively, copy all of its styles to a class specific to the Social Link. These guidelines are meant to ensure that styles only apply within the context of a component. Otherwise, we risk future breakage on changes expected only to affect `URLInput`. 

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

>A component's class name should **never** be used outside its own folder (with rare exceptions such as `_z-index.scss`). If you need to inherit styles of another component in your own components, you should render an instance of that other component. At worst, you should duplicate the styles within your own component's stylesheet. This is intended to improve maintainability by treating individual components as the isolated abstract interface.

aduth commented:
Was it intentional or discussed anywhere that this block be implemented as a [dynamic block](https://developer.wordpress.org/block-editor/tutorials/block-tutorial/creating-dynamic-blocks/), as opposed to strictly a static block with a `save` function to render its saved output? It doesn't seem to me that this block needs to be generated dynamically, as the server-side implementation appears to me to be generating only static content.

nosolosw commented:
@aduth the reason is that lower-privilege users aren't allowed to add SVG to `post_content`.

aduth commented:
@nosolosw Thanks, I can see how that might be problematic. But at least under that line of thinking, I would still consider it to be static content, ideally not a dynamic block. I might wonder if it's really a security issue to allow a contributor to publish content with `svg` (i.e. maybe we should loosen that restriction).

I did also edit my original comment in the meantime, since I do think there could be argument made that these _are_ technically dynamic content, if you consider that a social media provider may choose to update their logo at some point in the future, and by implementing this as a dynamic block, we can easily update all instances of that block in previously-published content.

mcsf commented:
> there could be argument made that these _are_ technically dynamic content, if you consider that a social media provider may choose to update their logo at some point in the future, and by implementing this as a dynamic block, we can easily update all instances of that block in previously-published content.

More or less rehashing arguments from Slack: technically, the essential content of the block is the reference to the social media provider (e.g. `service: 'spotify'`); the actual logo is derived data that can and should be pushed out of the static blocks (and, say, moved to stylesheets, which are subject to change).

mkaz commented:
I do think allowing the SVG content would be the best possible solution. However, here's a good presentation explaining the dangers of SVG and why the content can't be trusted: https://www.owasp.org/images/0/03/Mario_Heiderich_OWASP_Sweden_The_image_that_called_me.pdf

My thought would be to limit the block to Admins only, and restrict it from showing up to users without the privilege. This would be a nice feature regardless of this block.

I do have a PR here that moves the icons out to CSS, this would allow us to make it not dynamic https://github.com/WordPress/gutenberg/pull/19888

mtias commented:
Main reason was avoiding serializing SVG markup in the source, which would make any updates to logos unfeasible. The better solution would be to generate the SVGs through CSS so that markup can still be static while the icons can be iterated by updating the stylesheet. The support for coloring and such wasn't great at the time.
